File: javascript/bob/bob.js
var Bob = function() {};

Bob.prototype.hey = function(input) {
  if (input.toLowerCase() != input && input.toUpperCase() === input) {
    return 'Whoa, chill out!'
  } else if (input.slice(-1) === '?') {
    return 'Sure.'
  } else if (!input.trim()) {
    return 'Fine. Be that way!'
  } else {
    return 'Whatever.'
  }
};

module.exports = Bob;

File: elixir/bob/bob.exs
defmodule Teenager do
  def hey(string) do
    cond do
      String.ends_with?(string, "?") ->
        "Sure."
      String.strip(string) == "" ->
        "Fine. Be that way!"
      String.upcase(string) == string && String.match?(string, ~r/\p{L}+/) ->
        "Whoa, chill out!"
      true ->
        "Whatever."
    end
  end
end
